ddtrace/_trace/tracer.py

```diff
@@ -773,10 +773,8 @@ def _start_span(
 
         # Update the service name based on any mapping
         service = config.service_mapping.get(service, service)
-
         links = context._span_links if not parent else []
-
-        if trace_id:
+        if trace_id or links:
```

Comment:
Since we now count just span_links as a non-empty context, we need to make sure we initialize the span with them if they're available. This seems a little worrisome because we rely on the span __init__ to make the right choices when we pass in None for some of these values. However, it does work.